src/hotspot/os/linux/cgroupV2Subsystem_linux.cpp

```diff
@@ -244,3 +244,19 @@ char* CgroupV2Controller::construct_path(char* mount_path, char *cgroup_path) {
   return os::strdup(buf);
 }
 
+jlong CgroupV2Subsystem::pids_max() {
+  // we have to handle the special "max" value
+  GET_CONTAINER_INFO(jlong, _unified, "/pids.max",
+                     "Maximum number of tasks is: " JLONG_FORMAT, JLONG_FORMAT, pidsmax);
+  // not a number -> could be "max"
+  if (pidsmax < 0) {
+    char myline[1024];
+    int err2;
+    err2 = subsystem_file_line_contents(_unified, "/pids.max", NULL, "%1023s", myline);
+    if (err2 != 0) {
+      if (strncmp(myline, "max", 3) == 0) return -3;
+    }
+  }
+  return pidsmax;
```

Comment:
We have this pattern of needing to handle `max` elsewhere in cgroups v2 code. See for example: `CgroupV2Subsystem::cpu_quota()`. We should handle it similarly here.